src/frontend/src/lib/stores/i18n.store.ts

```diff
@@ -22,95 +12,39 @@ export const enI18n = (): I18n => ({
-});
-
-const deI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: de as I18n }),
-	lang: Languages.GERMAN
-});
+const loadLangI18n = async (lang: Languages): Promise<I18n> => {
+	const langMod = await import(`$lib/i18n/${lang}.json`);
 
-const frI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: fr as I18n }),
-	lang: Languages.FRENCH
-});
-
-const hiI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: hi as I18n }),
-	lang: Languages.HINDI
-});
+	const targetLang = langMod.default as I18n;
 
-const itI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: it as I18n }),
-	lang: Languages.ITALIAN
-});
-
-const jaI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: ja as I18n }),
-	lang: Languages.JAPANESE
-});
-
-const plI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: pl as I18n }),
-	lang: Languages.POLISH
-});
-
-const ptI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: pt as I18n }),
-	lang: Languages.PORTUGUESE
-});
-
-const viI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: vi as I18n }),
-	lang: Languages.VIETNAMESE
-});
-
-const zhcnI18n = (): I18n => ({
-	...mergeWithFallback({ refLang: enI18n(), targetLang: zhcn as I18n }),
-	lang: Languages.CHINESE_SIMPLIFIED
-});
+	return {
+		...mergeWithFallback({ refLang: enI18n(), targetLang }),
+		lang
+	};
+};
 
-const loadLang = (lang: Languages): I18n => {
-	switch (lang) {
-		case Languages.CHINESE_SIMPLIFIED:
-			return zhcnI18n();
-		case Languages.CZECH:
-			return csI18n();
-		case Languages.FRENCH:
-			return frI18n();
-		case Languages.GERMAN:
-			return deI18n();
-		case Languages.HINDI:
-			return hiI18n();
-		case Languages.ITALIAN:
-			return itI18n();
-		case Languages.JAPANESE:
-			return jaI18n();
-		case Languages.POLISH:
-			return plI18n();
-		case Languages.PORTUGUESE:
-			return ptI18n();
-		case Languages.VIETNAMESE:
-			return viI18n();
-		default:
-			return enI18n();
+const loadLang = async (lang: Languages): Promise<I18n> => {
+	try {
+		return await loadLangI18n(lang);
+	} catch (_: unknown) {
+		return enI18n();
 	}
 };
 
 const saveLang = (lang: Languages) => set({ key: 'lang', value: lang });
 
 export interface I18nStore extends Readable<I18n> {
-	init: () => void;
-	switchLang: (lang: Languages) => void;
+	init: () => Promise<void>;
+	switchLang: (lang: Languages) => Promise<void>;
 }
 
 const initI18n = (): I18nStore => {
-	const { subscribe, set } = writable<I18n>(loadLang(getDefaultLang()));
+	const { subscribe, set } = writable<I18n>(enI18n());
+
+	const switchLang = async ({ lang, track = true }: { lang: Languages; track?: boolean }) => {
+		const language = await loadLang(lang);
 
-	const switchLang = ({ lang, track = true }: { lang: Languages; track?: boolean }) => {
-		set(loadLang(lang));
+		set(language);
 
 		if (track) {
```

Comment:
Unrelated: is it expected to have an analytics event within the store? Maybe opiniated but, it makes the store impure which is a bit meh from my point of view.